src/pcm.c

```diff
@@ -1531,20 +1546,25 @@ static int pcm_mmap_transfer(struct pcm *pcm, void *buffer, unsigned int frames)
     }
 
     while (frames) {
+	pcm_sync_ptr(pcm, 0);
         avail = pcm_mmap_avail(pcm);
 
-        if (!avail) {
+        if (avail < pcm->config.avail_min) {
+            int time = -1;
             if (pcm->flags & PCM_NONBLOCK) {
                 errno = EAGAIN;
                 break;
             }
+	    if (pcm->flags & PCM_NOIRQ)
+                time = (pcm->config.avail_min - avail) / pcm->noirq_frames_per_msec;
```

Comment:
The `time` might be zero when the avail_min is close to avail. Is that expected?

Reply:
If the time is zero, it will return without waiting, then continue with the while loop from here to check if the data is available to write. it's expected.
https://github.com/tinyalsa/tinyalsa/pull/259/files#diff-3f4d00bbfd80c7c121a953205cb538e67d28f3a9a1c017e4b6e5fd66ee038afeR1549

Reply:
I'm not quite sure whether this makes the CPU really busy. Is it possible to set the time as 1 if it is zero?

Reply:
The time should not be zero, because pcm->config.avail_min is always smaller than avail here. but it might be close to avail and the time will be very small. do you mean if 0<time<1, we can set the time to 1.
such as:
if (time > 0 && time < 1) {
    time = 1;
}

Reply:
Yes, I mean if (pcm->config.avail_min - avail) is less than pcm->noirq_frames_per_msec

Reply:
yes, we can make sure the wait time is at least 1ms.